src/apm-cli.js

```diff
@@ -195,66 +195,69 @@ var getPythonVersion = function(callback) {
 };
 
 module.exports = {
-  run(args, callback) {
-    let Command;
-    config.setupApmRcFile();
-    const options = parseOptions(args);
+  run(args) {
+    return new Promise((resolve, reject) => {
+      let Command;
+      config.setupApmRcFile();
+      const options = parseOptions(args);
 
-    if (!options.argv.color) {
-      colors.disable();
-    }
+      if (!options.argv.color) {
+        colors.disable();
+      }
 
-    let callbackCalled = false;
-    options.callback = function(error) {
-      if (callbackCalled) { return; }
-      callbackCalled = true;
-      if (error != null) {
-        let message;
-        if (_.isString(error)) {
-          message = error;
-        } else {
-          message = error.message != null ? error.message : error;
-        }
+      let callbackCalled = false;
+      options.callback = function(error) {
+        if (callbackCalled) { return; }
+        callbackCalled = true;
+        if (error != null) {
+          let message;
+          if (_.isString(error)) {
+            message = error;
+          } else {
+            message = error.message != null ? error.message : error;
+          }
 
-        if (message === 'canceled') {
-          // A prompt was canceled so just log an empty line
-          console.log();
-        } else if (message) {
-          console.error(message.red);
+          if (message === 'canceled') {
+            // A prompt was canceled so just log an empty line
+            console.log();
+          } else if (message) {
+            console.error(message.red);
+          }
+          return void reject(error);
```

Comment:
I'm pretty sure this is somewhat equivalent to `return undefined`. Unless the side effect of running `reject(error)` is all you care about here -- if the resulting value of `reject(error)` needs to genuinely be `return`ed, then `void` should not be used here.

I haven't seen `void` used in a JS context before, that I can recall, so I had to look it up:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void

(My thoughts on `void` in JS: Seems kinda funky, IMO. Dunno when it'd be needed.)

---

No time for a full-on review at the moment, this just stood out to me and I thought I'd comment on it.

Reply:
Yes, the side effect is all I care about here. Basically `reject(error); return;`, or from a different approach: returning right after a function call while preventing the value to leak out.